docs/ReleaseProcess.md

```diff
@@ -350,3 +308,99 @@ git pull
 
 [contributing-to-a-project]: https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#Commit-Guidelines
 [chap]: https://github.com/input-output-hk/cardano-haskell-packages
+
+# The first time we had to break this release process
+
+On the first half of April 2023, we were asked to get a new Consensus release
+from current `master` in order for it to be integrated into the node. However
+the version on `master` could not be released because it was dependent on some
+changes from network that were not yet released.
+
+In order to make this clear, I am adding a `+` at the end of the version number
+to indicate that it contains unreleased changes on top of the version mentioned
+(which would be in CHaP). Note that in the cabal version, the `+` is not present
+so for cabal `0.3.1.0+` and `0.3.1.0` are both `0.3.1.0`.
+
+The release engineer that was integrating the node at that time, had pinned a
+specific commit in a `source-repository-packages` clause in his integration
+branch but by luck, several circumstances took place:
+- the commit was in `0.3.1.0+`
+- only `ouroboros-consensus-diffusion` depended on the network changes
+- `ouroboros-consensus` with his changes was, as mentioned above, at `0.3.1.0+`
+- `ouroboros-consensus-diffusion 0.3.1.0+` would not have been buildable at the
+  current CHap versions
+- but `ouroboros-consensus-diffusion 0.3.1.0` (the one from CHaP) declared a
+  dependency on `ouroboros-consensus 0.3.1.0`
+- and by chance, the changes in `ouroboros-consensus 0.3.1.0+` (which actually
+  were "Breaking changes") didn't break the compilation of
+  `ouroboros-consensus-diffusion 0.3.1.0`
+- so he could pull `ouroboros-consensus 0.3.1.0+` from the
+  `source-repository-packages` and `ouroboros-consensus-diffusion 0.3.1.0` from
+  CHaP
+
+So in that situation we were breaking the bundle in a way that our previous
+rules couldn't handle.
+
+> Problem 1: We were unable to make a release from `master`, so we had to make
+> the release from some previous commit as a side branch.
+
+> Problem 2: Our bundles were too rigid for the development that happens.
```

Comment:
I'm not sure what this sentence means. I don't know if it could be expanded somehow.